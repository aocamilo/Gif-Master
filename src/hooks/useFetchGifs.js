import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: false,
  });

  useEffect(() => {
    setState({ ...state, loading: true });
    getGifs(category).then((imgs) => {
      setState({
        loading: false,
        data: imgs,
      });
    });
  }, [category]);

  return state;
};
