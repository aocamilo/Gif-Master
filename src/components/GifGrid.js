import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);

  // useEffect(() => {
  //   getGifs(category).then(setImages);
  // }, [category]);

  const { data: images, loading } = useFetchGifs(category);

  if (loading)
    return <p className='animate__animated animate__flash'>Loading...</p>;
  return (
    <>
      <h3> {category} </h3>
      <div className='card-grid'>
        {images.length > 0 &&
          images.map((img) => <GifGridItem key={img.id} {...img} />)}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
