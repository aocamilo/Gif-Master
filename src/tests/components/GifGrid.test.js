import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); //Mock todo lo que llamen de esta ruta

describe('Pruebas a <GifGrid />', () => {
  const category = 'Naruto';

  test('should show the correct component and match with the snapshot', () => {
    useFetchGifs.mockReturnValue({
      //Mock el return cuando se llame el hook
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar items cuando se cargan imagenes usando useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/cualquier/cosa.jpg',
        title: 'Cualquier cosa',
      },
      {
        id: 'ABD',
        url: 'https://localhost/cualquier/cosa.jpg',
        title: 'Cualquier cosa',
      },
    ];

    useFetchGifs.mockReturnValue({
      //Mock el return cuando se llame el hook
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot(); // Prueba que el snapshot se haya actualizado
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length); //Search a custom component
  });
});
