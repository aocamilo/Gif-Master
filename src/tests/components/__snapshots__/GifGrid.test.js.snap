// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifGrid /> Debe mostrar items cuando se cargan imagenes usando useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    Naruto
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="ABD"
      key="ABD"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas a <GifGrid /> should show the correct component and match with the snapshot 1`] = `
<p
  className="animate__animated animate__flash"
>
  Loading...
</p>
`;
